### DISCUSSION TITLE
How to Set Up a Simple Query in TypeGraphQL?

### DISCUSSION THREAD

**Discussion by avocado1109** [MichalLytek/type-graphql#1750] [Q&A] [answered]
I'm new to TypeGraphQL and trying to set up a simple query to fetch user data from a mock array of users. I'm not sure how to structure the query or the resolver properly. Can someone provide a basic example of how to create a query in TypeGraphQL that fetches a list of users?

**Comment by MichalLytek** [comment]
https://github.com/MichalLytek/type-graphql/blob/master/examples/simple-usage/recipe.resolver.ts

**Comment by skyward06** [comment, marked as answer]
Setting up a simple query in TypeGraphQL is straightforward. Here's an example where we define a User type and a query to return a list of users from a mock array:

```ts
import "reflect-metadata";
import { ObjectType, Field, Int, Query, Resolver } from "type-graphql";

@ObjectType()
class User {
  @Field(() => Int)
  id: number;

  @Field()
  name: string;

  @Field()
  email: string;
}

const mockUsers = [
  { id: 1, name: "John Doe", email: "john@example.com" },
  { id: 2, name: "Jane Smith", email: "jane@example.com" },
];

@Resolver()
class UserResolver {
  @Query(() => [User])
  users() {
    return mockUsers;
  }
}
```
